Linux build failure

**Describe the bug**
Warning on always-true component of assertion causes build failure with GCC 10.1.0.

**To Reproduce**
1. Clone repository
2. Initialize submodules
3. Run cmake
4. Run make.

**Expected behavior**
Build success

**Desktop (please complete the following information):**
 - OS: Linux
 - GPU: N/A for this issue
 - CPU: N/A for this issue (x86_64)
 - Version: 59839ce06a06742e31689f319da3dbc4f635bc18

**Additional context**
Changing the line to `assert(index < 6);` resolves the issue.

**Log file**
Irrelevant path components have been replaced with `(...)` in this the log excerpt.
```
[ 82%] Building CXX object Main/CMakeFiles/usc-game.dir/src/Scoring.cpp.o
In file included from /usr/include/c++/10.1.0/cassert:44,
                 from (...)/unnamed-sdvx-clone/third_party/nlohmann_json/json.hpp:10910,
                 from (...)/unnamed-sdvx-clone/Main/include/Game.hpp:5,
                 from (...)/unnamed-sdvx-clone/Main/include/Scoring.hpp:5,
                 from (...)/unnamed-sdvx-clone/Main/src/Scoring.cpp:2:
(...)/unnamed-sdvx-clone/Main/src/Scoring.cpp: In member function ‘bool Scoring::m_IsBeingHold(const ScoreTick*) const’:
(...)/unnamed-sdvx-clone/Main/src/Scoring.cpp:995:11: error: comparison of unsigned expression in ‘>= 0’ is always true [-Werror=type-limits]
  995 |  assert(0 <= index && index < 6);
      |         ~~^~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [Main/CMakeFiles/usc-game.dir/build.make:381: Main/CMakeFiles/usc-game.dir/src/Scoring.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:741: Main/CMakeFiles/usc-game.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
```

--- a/Main/src/Scoring.cpp
+++ b/Main/src/Scoring.cpp
@@ -989,7 +989,7 @@ bool Scoring::m_IsBeingHold(const ScoreTick* tick) const
 
 	const HoldObjectState* obj = (HoldObjectState*) tick->object;
 	const uint32 index = obj->index;
-	assert(0 <= index && index < 6);
+	assert(index < 6);
 	
 	// Button needs to be hold at this moment.
 	// (Unless `tick` is the end of a long note; see below)

